FamilyTreeTesting/Scripts/ImprovedScript/DataModel.js

```diff
@@ -163,6 +109,25 @@ function searchParentTreeNodePreviousNode(currentBranchNode, inputId) {
     return resultNode;
 }
 
+function searchNodeCurrentArray(childrenList, inputId) {
+    var resultListData = [];
+    childrenList.forEach(function(child,index){
+        if (child.id === inputId){
+            resultListData.push(childrenList);
+            resultListData.push(index);
```

Comment:
this should be
result.resultChildrenList = childrenList;
result.resultIndex = index;